core/domain/exp_services.py

```diff
@@ -1795,3 +1799,203 @@ def save_multi_exploration_math_rich_text_info_model(
+    if old_state_dict['interaction']['hints'] != (
+            new_state_dict['interaction']['hints']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_HINTS,
+                new_state_dict['interaction']['hints']))
+    if old_state_dict['interaction']['solution'] != (
+            new_state_dict['interaction']['solution']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION,
+                new_state_dict['interaction']['solution']))
+    if old_state_dict['interaction']['answer_groups'] != (
+            new_state_dict['interaction']['answer_groups']):
+        property_name_to_new_value_mapping.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS,
+                new_state_dict['interaction']['answer_groups']))
+
+    for property_name, new_value in property_name_to_new_value_mapping:
+        change_lists.append(
+            exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'state_name': state_name,
+                'property_name': property_name,
+                'new_value': new_value
+            }))
+    return change_lists
+
+
+def get_latex_strings_and_exp_ids_for_generating_svgs():
+    """Returns a batch of LaTeX strings from explorations which have LaTeX
+    strings without SVGs.
+
+    Returns:
+        dict(str, list). The dict having each key as an exp_id and value
+        as a list of LaTeX string.
+    """
+
+    latex_strings_mapping = {}
+    exploration_math_rich_text_info_models = (
+        exp_models.ExplorationMathRichTextInfoModel.get_all())
+    size_of_svgs_in_batch = 0
+    for model in exploration_math_rich_text_info_models:
+        if model.math_images_generation_required:
+            latex_strings_mapping[model.id] = model.latex_strings_without_svg
+            size_of_svgs_in_batch += (
+                model.estimated_max_size_of_images_in_bytes)
+            if size_of_svgs_in_batch > feconf.MAX_SIZE_OF_MATH_SVGS_BATCH_BYTES:
+                return latex_strings_mapping
+    return latex_strings_mapping
+
+
+def update_explorations_with_math_svgs(user_id, exp_id, image_data):
+    """Saves an SVG for each latex value without an SVG in an exploration
+    and updates the exploration. Also the corresponding valid draft changes are
+    updated.
+
+    Args:
+        image_data: dict. The dictionary having all the image data like latex
+            value and dimensions which will be used for generating and assigning
+            filenames.
+        user_id: str. The ID of the commiter.
+        exp_id: str. The ID of the exploration to update.
+
+    Raises:
+        Exception: If any of the SVG images provided fail validation.
+    """
+    exploration = exp_fetchers.get_exploration_by_id(exp_id)
+    html_strings_after_conversion = ''
```

Comment:
The plural indicates a list. Try to pick a better name to explain what exactly this is.

Reply:
done